rest/route/service.go

```diff
@@ -65,6 +65,7 @@ func AttachHandler(app *gimlet.APIApp, queue amboy.Queue, URL string, superUsers
 	app.AddRoute("/patches/{patch_id}/abort").Version(2).Post().Wrap(checkUser).RouteHandler(makeAbortPatch(sc))
 	app.AddRoute("/patches/{patch_id}/restart").Version(2).Post().Wrap(checkUser).RouteHandler(makeRestartPatch(sc))
 	app.AddRoute("/projects").Version(2).Get().RouteHandler(makeFetchProjectsRoute(sc))
+	app.AddRoute("/projects/{project_id}/events").Version(2).Get().Wrap(checkUser).RouteHandler(makeFetchProjectEvents(sc))
```

Comment:
I think there's a checkProject or projectContext middleware that we want to wrap here.

Reply:
It looks to me that the projectContext middleware lets the user through if a) the user is authenticated or b) the project is not private. Is that better than checkUser which just lets an authenticated user through? 
checkProject looks like it's used in RestV1.

Reply:
If I make the changes John suggested I can remove the middleware wrappers since the handler will check the user is authorized

Reply:
I think we should keep the wrappers, and I think you want both projectContext *and* user. I'd tend to prefer things on the side of using the wrappers over just letting routes reimplement things.

might make sense to make middlware to filter out projectAdmin operations.

Reply:
I made a new middleware to handle it and removed the check from the route